File: api/app/Models/Warehouse.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class Warehouse extends Model
{
    use SoftDeletes;
    protected $fillable = ['name', 'warehouse_type_id', 'code', 'address', 'type', 'city', 'is_active', 'is_pos'];

    protected $casts = [
        'is_active' => 'boolean',
        'is_pos' => 'boolean',
    ];

    public function warehouseType(): BelongsTo
    {
        return $this->belongsTo(WarehouseType::class);
    }

    public function stocks(): HasMany
    {
        return $this->hasMany(Stock::class, 'warehouse_id');
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Catalog/Entities/Warehouse.cs
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Inventory.Entities;
using AdminHiitop.Api.Domain.Sales.Entities;

namespace AdminHiitop.Api.Domain.Catalog.Entities;

public sealed class Warehouse : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public int? WarehouseTypeId { get; set; }
    public string? Type { get; set; }
    public string? City { get; set; }
    public int? ProvinceId { get; set; }
    public int? DistrictId { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsPos { get; set; }

    public WarehouseType? WarehouseType { get; set; }
    public Province? Province { get; set; }
    public District? District { get; set; }
    public ICollection<Stock> Stocks { get; set; } = new List<Stock>();
    public ICollection<Order> Orders { get; set; } = new List<Order>();
    public ICollection<StockMovement> StockMovements { get; set; } = new List<StockMovement>();
}
